assets/js/base/context/utils.js

```diff
@@ -0,0 +1,62 @@
+/**
+ * This is an assertion utility for validating that the incoming value prop
+ * value on a given context provider is valid and throws an error if it isn't.
+ *
+ * Note: this asserts values that are expected to be an object.
+ *
+ * The validationMap is expected to be an object in the following shape.
+ *
+ * {
+ *   [expectedPropertyName<String>]: {
+ *     required: [expectedRequired<Boolean>]
+ *     type: [expectedtype<String>]
+ *   }
+ * }
+ *
+ * @param {string} contextName   The name of the context provider being
+ *                               validated.
+ * @param {Object} validationMap A map for validating the incoming value against.
+ * @param {Object} value         The value being validated.
+ *
+ * @throws {Error}
+ */
+export const assertValidContextValue = (
+	contextName,
+	validationMap,
+	value
+) => {
+	if ( typeof value !== 'object' ) {
+		throw new Error(
+			`${ contextName } expects an object for its context value`
+		);
+	}
+	const errors = [];
+	for ( const expectedProperty in validationMap ) {
+		if (
+			validationMap[ expectedProperty ].required &&
+			typeof value[ expectedProperty ] === 'undefined'
+		) {
+			errors.push(
+				`The ${ expectedProperty } is required and is not present.`
+			);
```

Comment:
Should we add a `continue` here? If the expected property is not present, there is no need to do the validation below (it will never get inside the `if` statement).

Another option would be to replace the second `if` with and `else if`.

Reply:
good catch